test/aircraft/FlightManagementSystem/Fms.spec.js

```diff
@@ -819,6 +820,74 @@ ava('._prependLegCollectionWithRouteAmendment() adds LegModels for each routeStr
     t.true(fms.legCollection[1].routeString === routeAmmendmentMock[1]);
 });
 
+ava('._regenerateSidLeg() creates a new SID leg of identical route string for the currently assigned departure runway', (t) => {
```

Comment:
be careful with multiple matchers (`t.true`, `t.false`, etc) in a single test.  In the event one of them fails you will not get any indication on which one, specifically, failed.  

try to keep tests small as possible.  this is a big one, though I think it could be shorter.  see below